woocommerce/woocommerce-android
# Issue/12894 blaze cta text input

Closes: #12894 

⚠️ Depends on [FluxC changes](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3108)


### Description

Adds a new input text field in the Blaze campaign creation flow to edit the CTA text of the ad. It also updates the style of the ad CTA to match the [Blaze ad style.](https://github.com/woocommerce/woocommerce-android/pull/12899#issuecomment-2467710004)

### Testing information
1. Log into a site with Blaze enabled
2. Trigger Blaze campaign creation flow
3. Once suggestions are loaded, click on `Edit Ad` button
4. Check the input text "Call to action" is prefilled for all 3 suggestions. 
5. Edit the value of it and save it
6. Check the ad preview is updated with the edited value
7. Create the campaign and load the campaign detail review. Verify the ad CTA shows the value you entered. 

### The tests that have been performed

The above

### Images/gif


https://github.com/user-attachments/assets/990428cf-36bc-4ee0-a1f8-4f89add7a9cb

Changes applied to ad CTA: 


https://github.com/user-attachments/assets/d6fd2d68-ee72-459d-b15a-cfc0cb4bfcaa


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
